Content/build.fsx

```diff
@@ -113,8 +112,71 @@ Target "Docker" (fun _ ->
   let tagArgs = sprintf "tag %s %s" dockerFullName dockerFullName
   run "docker" tagArgs "."
 )
+
 #endif
+#if (Azure)
+Target "Publish" (fun () ->
+  run yarnTool "install --frozen-lockfile" __SOURCE_DIRECTORY__
+  run dotnetCli (sprintf "publish %s -c release -o %s" serverPath deployDir) __SOURCE_DIRECTORY__
+  run dotnetCli "restore" clientPath
+  run dotnetCli "fable webpack -- -p" clientPath
+  CopyDir (deployDir </> "public") (clientPath </> "public") allFiles
+)
 
+type ArmOutput =
+  { WebAppName : ParameterValue<string>
+    WebAppPassword : ParameterValue<string> }
+let environment = getBuildParamOrDefault "environment" (Guid.NewGuid().ToString().ToLower().Split '-' |> Array.head)
+let subscriptionId = getBuildParam "subscriptionId"
+let clientId = getBuildParam "clientId"
+
+let mutable deploymentOutputs : ArmOutput option = None
+
+Target "DeployArmTemplate" (fun _ ->
+  let armTemplate = @"arm-template.json"
+  let resourceGroupName = "safe-" + environment
+  let subscriptionId = Guid.Parse subscriptionId
+  let clientId = Guid.Parse clientId
+
+  tracefn "Deploying template '%s' to resource group '%s' in subscription '%O'..." armTemplate resourceGroupName subscriptionId
+
+  let authCtx =
+    subscriptionId
+    |> authenticateDevice trace { ClientId = clientId; TenantId = None }
+    |> Async.RunSynchronously
+
+  let deployment =
+     { DeploymentName = "fake-deploy"
+       ResourceGroup = New(resourceGroupName, Region.EuropeWest)
+       ArmTemplate = IO.File.ReadAllText armTemplate
+       Parameters = Simple [ "environment", ArmString environment ]
+       DeploymentMode = Incremental }
+
+  deployment
+  |> deployWithProgress authCtx
+  |> Seq.iter(function
+    | DeploymentInProgress (state, operations) -> tracefn "State is %s, completed %d operations." state operations
+    | DeploymentError (statusCode, message) -> traceError <| sprintf "DEPLOYMENT ERROR: %s - '%s'" statusCode message
+    | DeploymentCompleted d -> deploymentOutputs <- d)
+)
+
+Target "DeployWebApp" (fun _ ->
+  let zipFile = "deploy.zip"
+  IO.File.Delete zipFile
+  Zip deployDir zipFile !!(deployDir + @"\**\**")
+  
+  let appName = deploymentOutputs.Value.WebAppName.value
+  let appPassword = deploymentOutputs.Value.WebAppPassword.value
+  let destinationUri = sprintf "https://%s.scm.azurewebsites.net/api/zipdeploy" appName
+  tracefn "Uploading %s to %s" zipFile destinationUri
+  let client = new Net.WebClient(Credentials = Net.NetworkCredential("$" + appName, appPassword))
+  client.UploadData(destinationUri, IO.File.ReadAllBytes zipFile) |> ignore
+)
+
+"Publish" ==> "DeployWebApp"
```

Comment:
Can we put that into a single chain?

Reply:
Technically this is what the dependencies are (the ARM deploy only sets up Azure infrastructure) but @isaacabraham has already put them in a single chain and I'm happy with it.